tsl/src/bgw_policy/job.c

```diff
@@ -379,6 +381,92 @@ bgw_policy_job_check_enterprise_license(BgwJob *job)
 	return required;
 }
 
+static void
+job_execute_function(FuncExpr *funcexpr)
+{
+	bool isnull;
+
+	EState *estate = CreateExecutorState();
+	ExprContext *econtext = CreateExprContext(estate);
+
+	ExprState *es = ExecPrepareExpr((Expr *) funcexpr, estate);
+	ExecEvalExpr(es, econtext, &isnull);
+
+	FreeExprContext(econtext, true);
+	FreeExecutorState(estate);
+}
+
+static void
+job_execute_procedure(FuncExpr *funcexpr)
+{
+	CallStmt *call = makeNode(CallStmt);
+	call->funcexpr = funcexpr;
+	DestReceiver *dest = CreateDestReceiver(DestNone);
+	/* we don't need to create proper param list cause we pass in all arguments as Const */
+#ifdef PG11
+	ParamListInfo params = palloc0(offsetof(ParamListInfoData, params));
+#else
+	ParamListInfo params = makeParamList(0);
+#endif
+	ExecuteCallStmt(call, params, false, dest);
+}
+
+static bool
```

Comment:
What's the reason to return `bool` here and what does it mean? It seems the result is always `true`.

I guess some kind of exit code from the user function would make sense, but that's not the case here.

Reply:
i was just adhering to the other execute_functions, it doesnt make much sense and should be removed once all the policies have been migrated, all the errors are handled by exceptions. It means the job run successfully.

This is all wrapped in try/catch in the scheduler so we dont need the return value

Reply:
I guess the reason for the bool is if there is any error condition we detect without throwing error we could signal it that way